src/components/TodoItem/index.tsx

```diff
@@ -0,0 +1,163 @@
+          onClick={(e) => {
+            e.stopPropagation();
+            setIsNoteDetailOpen(true);
+          }}
+          aria-label="View note"
+        >
+          <NoteViewIcon width={24} height={24} />
+        </button>
+      ) : (
+        <button
+          type="button"
+          className="flex h-6 w-6 flex-shrink-0 items-center justify-center rounded-full bg-slate-50"
+          onClick={(e) => {
+            e.stopPropagation();
+            navigate('/notes/new', { state: todo });
+          }}
+          aria-label="Write note"
+        >
+          <NoteWriteIcon width={24} height={24} />
+        </button>
+      )}
+    </div>
+  );
+}
+
+function TodoItem({
+  todo,
+  showGoals = false,
+  showIcons = false,
+  hoverBgColor = 'hover:bg-slate-100',
+}: TodoItemProps) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isNoteDetailOpen, setIsNoteDetailOpen] = useState(false);
+  const navigate = useNavigate();
+  const { mutate: toggleDone } = usePatchDone();
+
+  return (
+    <>
+      <div
+        key={todo.id}
+        className={`group relative flex cursor-pointer items-center justify-between rounded-lg px-1 py-[2px] transition-all duration-200 ${hoverBgColor}`}
+        onClick={() => setIsModalOpen(true)}
+        role="button"
+        tabIndex={0}
+        onKeyDown={() => setIsModalOpen(true)}
+      >
+        <div className="flex w-full items-start space-x-2 overflow-hidden">
+          <button
+            type="button"
+            className="flex h-6 w-6 flex-shrink-0 items-center justify-center"
+            onClick={(e) => {
+              e.stopPropagation();
+              toggleDone(todo);
+            }}
+            aria-label={todo.done ? 'Mark as incomplete' : 'Mark as complete'}
+          >
+            {todo.done ? (
+              <ActiveBlue width={24} height={24} />
+            ) : (
+              <Inactive width={24} height={24} />
+            )}
+          </button>
+          <div className="min-w-0 flex-1 items-center overflow-hidden">
+            <div
+              className={`pt-[1px] text-sm font-normal leading-6 text-slate-800 ${todo.done ? 'line-through' : ''} truncate whitespace-nowrap`}
+            >
+              {todo.title}
+            </div>
+            {showGoals && todo.goal !== null && (
+              <div className="flex items-center overflow-hidden">
+                <div className="flex h-6 w-6 flex-shrink-0 items-center justify-center rounded-full bg-slate-50">
+                  <GoalIcon width={24} height={24} />
+                </div>
+                <div className="ml-2 truncate text-sm font-normal text-slate-700">
+                  {todo.goal.title}
+                </div>
+              </div>
+            )}
+          </div>
+        </div>
+        {showIcons && (
+          <IconButtons
+            todo={todo}
+            setIsNoteDetailOpen={setIsNoteDetailOpen}
```

Comment:
```suggestion
            onNoteDetailOpen={()=>setIsNoteDetailOpen(true)}
```
[Suggestion] 이런식으로 한번 감싸서 보내는것도 안전해 보입니당